src/google/protobuf/stubs/port.h

```diff
@@ -263,7 +263,7 @@ static const uint64 kuint64max = GOOGLE_ULONGLONG(0xFFFFFFFFFFFFFFFF);
 # endif
 #endif
 
-#if GOOGLE_PROTOBUF_USE_UNALIGNED
+#ifdef GOOGLE_PROTOBUF_USE_UNALIGNED
```

Comment:
I think this would actually change the behavior. When we get here this macro should have been assigned a value of either 0 or 1, so we need to check the value of the macro and not whether it was defined. Does the warning disappear if you do something like this instead?
`#if defined(GOOGLE_PROTOBUF_USE_UNALIGNED) && GOOGLE_PROTOBUF_USE_UNALIGNED`